lib/Controller/ChatController.php

```diff
@@ -0,0 +1,198 @@
+	 *         found".
+	 */
+	public function sendMessage($token, $message) {
+		try {
+			$room = $this->manager->getRoomForParticipantByToken($token, $this->userId);
+		} catch (RoomNotFoundException $exception) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		if ($this->userId === null) {
+			$actorType = 'guests';
+			$actorId = $this->session->get('spreed-session');
+			// The character limit for actorId is 64, but the spreed-session is
+			// 256 characters long, so it has to be hashed to get an ID that
+			// fits (except if there is no session, as the actorId should be
+			// empty in that case but sha1('') would generate a hash too
+			// instead of returning an empty string).
+			$actorId = $actorId? sha1($actorId): $actorId;
+		} else {
+			$actorType = 'users';
+			$actorId = $this->userId;
+		}
+
+		if (!$actorId) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		$creationDateTime = new \DateTime('now', new \DateTimeZone('UTC'));
+
+		$this->chatManager->sendMessage($token, $actorType, $actorId, $message, $creationDateTime);
+
+		return new DataResponse([], Http::STATUS_CREATED);
+	}
+
+	/**
+	 * @PublicPage
+	 *
+	 * Receives the chat messages from the given room.
+	 *
+	 * It is possible to limit the returned messages to those not older than
+	 * certain date and time setting the $notOlderThan parameter. In the same
+	 * way it is possible to ignore the first N messages setting the $offset
+	 * parameter. Both parameters are optional; if not set all the messages from
+	 * the chat are returned.
+	 *
+	 * If there are currently no messages the response will not be sent
+	 * immediately. Instead, HTTP connection will be kept open waiting for new
+	 * messages to arrive and, when they do, then the response will be sent. The
+	 * connection will not be kept open indefinitely, though; the number of
+	 * seconds to wait for new messages to arrive can be set using the timeout
+	 * parameter; the default timeout is 30 seconds, maximum timeout is 60
+	 * seconds. If the timeout ends then a successful but empty response will be
+	 * sent.
+	 *
+	 * @param string $token the room token
+	 * @param int offset optional, the first N messages to ignore
+	 * @param int notOlderThanTimestamp optional, timestamp in seconds and UTC
+	 *        time zone
+	 * @param int timeout optional, the number of seconds to wait for new
+	 *        messages (30 by default, 60 at most)
+	 * @return DataResponse an array of chat messages, or "404 Not found" if the
+	 *         room token was not valid; each chat message is an array with
+	 *         fields 'id', 'token', 'actoryType', 'actorId',
+	 *         'actorDisplayName', 'timestamp' (in seconds and UTC timezone) and
+	 *         'message'.
+	 */
+	public function receiveMessages($token, $offset = 0, $notOlderThanTimestamp = 0, $timeout = 30) {
+		try {
+			$room = $this->manager->getRoomForParticipantByToken($token, $this->userId);
```

Comment:
We need to extend this, so it is only accessible when the user is listed, or has a valid session.
Otherwise you can read chat messages of password protected public rooms without the password.

So please use:
https://github.com/nextcloud/spreed/blob/86ec1dd446f8907e08c6c1d1380685b7b901a7fd/lib/Controller/CallController.php#L83-L99

Same for sending Messages

If that makes your integration tests fail try `And user "participant" joins call "room" with 200`